File: services/python/fastapi/src/routes/handlers/internal.py
import structlog
from fastapi import Request
from fastapi.responses import JSONResponse

from routes.errors import AppErrors, send_error


async def internal_handler(request: Request, exc: Exception) -> JSONResponse:
    structlog.get_logger().error(
        "Internal server error",
        method=request.method,
        path=request.url.path,
        error=repr(exc),
    )

    return send_error(AppErrors.Internal)

File: services/go/gin/routes/handlers/internal.go
package handlers

import (
	"gin/routes"
	"log/slog"

	"github.com/gin-gonic/gin"
)

func InternalHandler(c *gin.Context, err any) {
	slog.Error("Internal server error", "method", c.Request.Method, "path", c.Request.URL.Path, "error", err)

	routes.SendError(c, routes.AppErrors.Internal)
}
